lib/diaspora/mentions_container.rb

```diff
@@ -0,0 +1,41 @@
+module Diaspora
+  module MentionsContainer
+    extend ActiveSupport::Concern
+
+    included do
+      after_create :create_mentions
+      has_many :mentions, as: :mentions_container, dependent: :destroy
+    end
+
+    def mentioned_people
+      if persisted?
+        mentions.includes(person: :profile).map(&:person)
+      else
+        Diaspora::Mentionable.people_from_string(text)
+      end
+    end
+
+    def legitimate_mention_recipients
+      visible_for_recepient_mentions.map(&:person)
+    end
+
+    def visible_for_recepient_mentions
+      ppl = people_allowed_to_be_mentioned
+      if ppl == "all"
+        mentions
+      else
+        mentions.where(person: ppl)
```

Comment:
I'm not sure about this, this generates a `WHERE IN (...)` query with potentially many people (when someone with many contacts shares a post with "all aspects").